http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -404,6 +437,16 @@ object LeonardoJsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
     }
   }
 
+  implicit object JupyterDockerImageJsonFormat extends JsonFormat[JupyterDockerImage] {
+    def read(json: JsValue): JupyterDockerImage = json match {
+      case JsString(imageUrl) =>
+        JupyterDockerImage.stringToJupyterDockerImage(imageUrl)
+          .getOrElse(throw DeserializationException(s"Invalid docker registry. Only gcr.io or dockerhub are supported"))
```

Comment:
It may be more robust to use `allRegistry` values instead of hard-coding `gcr.io` and `dockerhub` in the exception message.